File: c++/diagonal-negativos.cpp
#include <iostream>
#include <iomanip>
#include <string>
#include <climits>
#include <cmath>

using namespace std;
int main() {

    int i, j, N, quant;

    quant = 0;

    cout << "Qual a ordem da matriz? ";
    cin >> N;

    int** mat = new int*[N];
    for (i = 0; i < N; i++)
        mat[i] = new int[N];

    for (i = 0; i < N; i++)
    {
        for (j = 0; j < N; j++)
        {
            cout << "Elemento [" << i << "," << j << "]: ";
            cin >> mat[i][j];
        }
        
    }

    for (i = 0; i < N; i++)
    {
        for (j = 0; j < N; j++)
        {
            if (mat[i][j] < 0)
            {
                quant++;
            }
            
        }
        
    }
    
    cout << "DIAGONAL PRINCIPAL:" << endl;

    for (i = 0; i < N; i++)
    {
        for (j = 0; j < N; j++)
        {
            if (i == j)
            {
                cout << mat[i][j] << " ";
            }
            
        }
    }

    cout << "\n";

    cout << "QUANTIDADE DE NEGATIVOS = " << quant << endl;

    return 0;
}

File: python/diagonal-negativos.py
N: int
i: int
j: int
quant: int

quant = 0

N = int(input("Qual a ordem da matriz? "))

mat: [[int]] = [[0 for x in range(N)] for x in range(N)]

for i in range(0, N):
    for j in range(0, N):
        mat[i][j] = int(input(f"Elemento [{i},{j}]: "))

for i in range(0, N):
    for j in range(0, N):
        if mat[i][j] < 0:
            quant = quant + 1

print("DIAGONAL PRINCIPAL: ")

for i in range(0, N):
    for j in range(0, N):
        if(i == j):
            print(f"{mat[i][j]}", end=" ")

print()
print(f"QUANTIDADE DE NEGATIVOS = {quant}")